src/controllers/profileController.ts

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable no-console */
+/* eslint-disable import/prefer-default-export */
+/* eslint-disable max-len */
+/* eslint-disable linebreak-style */
+import { Request, Response } from "express";
+import UserProfile from "../database/models/UserProfile";
+
+export const updateUserProfile = async (req: Request, res: Response) => {
+  try {
+    const userId = req.user.id;
```

Comment:
I think one model which is User is enough we don't need another to create another one which means better you can use User instead of creating new one (Profile) which can cause unnecessary duplication of data